browser/mapping.go

```diff
@@ -513,8 +515,10 @@ func mapPage(vu moduleVU, p *common.Page) mapping {
 		"mouse": rt.ToValue(p.GetMouse()).ToObject(rt),
 		"on": func(event string, handler goja.Callable) error {
 			vu.tqMu.Lock()
-			if vu.tq == nil {
-				vu.tq = taskqueue.New(vu.RegisterCallback)
+			tq := vu.tq[p.GetTargetID()]
+			if tq == nil {
+				tq = taskqueue.New(vu.RegisterCallback)
+				vu.tq[p.GetTargetID()] = tq
 			}
 			vu.tqMu.Unlock()
```

Comment:
I wonder if we should implement a higher level API in `moduleVU` to encapsulate this logic and similar ones that are in the mapping layer now so it is a little cleaner?

Reply:
Yep, that was something I was considering. Let me give it a go 👍

Reply:
Moved the implementation to a higher level (`taskqueueRegistry`): 8ddf4494b121cd895c155ecfcd128d8c6b72870e, fb900e1b26a7ee9f75cc668c9503ac5bdd597521, d422abaf4531c5e490fa876e62d8804287601580.